ml4gw/transforms/decimator.py

```diff
@@ -0,0 +1,137 @@
+import torch
+
+
+class Decimator(torch.nn.Module):
+    """
+    Downsample/decimate timeseries according to a schedule.
+
+    The schedule is a tensor of shape (N, 3), where each row is:
+    [start_time, end_time, target_sample_rate]
+
+    Example schedule (seconds, Hz):
+    [[0, 40, 256],   # first 40s at 256 Hz
+     [40, 58, 512],  # next 18s at 512 Hz
+     [58, 60, 2048]] # last 2s at full 2048 Hz
+
+    Strain: Tensor of shape (B, C, T), where
```

Comment:
I should've thought of this earlier: this transform is generically applicable to timeseries, so we should keep the variable names and documentation generic too. Could you go through and update the terminology? It's fine to use something like `X` for the data name, and just say something like "timeseries" instead of strain or waveforms.